openmicroscopy/openmicroscopy
# Bugfix: Import As freezes

Fixes [Ticket 12929](https://trac.openmicroscopy.org/ome/ticket/12929)
Problem: After an object has been saved the connector was shutdown. This could lead to an abortion of a parallel running import job. This PR removes the connector shutdown, because it's actually not necessary at that point anyway.

**Test**:
Follow the workflow which previously triggered this bug, which is:
- Log in as a group owner or administrator
- *Start*  (not just add to the queue) a long running import job for another user
- Now start another import for the user you're logged in with, but import into a particular dataset which you have to create first on the import dialog (this caused the crash)
- Make sure the imports are passing like expected.

Note: The actual bugfix is just commit 979098b . The first commits add PropertyChangeSupport to the Gateway, i. e. one can add a PropertyChangeListener to the Gateway for debugging purposes, which makes it much easier to monitor what's going on in the Gateway. Any objections on this?

/cc @jburel
